frontend/src/store/player-element.ts

```diff
@@ -132,51 +189,107 @@ class PlayerElementStore extends CommonStore<PlayerElementState> {
+       */
+      let applyJASSUB = !this._useCustomSubtitleTrack;
+
+      if (this._useCustomSubtitleTrack) {
+        const data = await genericWorker.parseSsaFile(subtitleTrack.src);
+
+        /**
+         * If style isn't basic (animations, custom typographics, etc.)
+         * fallback to rendering subtitles with JASSUB
+         */
+        if (data?.isBasic) {
+          this.currentExternalSubtitleTrack.parsed = data;
+        } else {
+          applyJASSUB = true;
+        }
+      }
+
+      if (applyJASSUB) {
+        const serverAddress = remote.sdk.api?.basePath;
+
+        const attachedFonts
+        = playbackManager.currentMediaSource?.MediaAttachments?.filter(a =>
+          this._isSupportedFont(a.MimeType)
+        )
+          .map((a) => {
+            if (a.DeliveryUrl && serverAddress) {
+              return `${serverAddress}${a.DeliveryUrl}`;
+            }
+          })
+          .filter((a): a is string => a !== undefined) ?? [];
+
+        this._setSsaTrack(subtitleTrack.src, attachedFonts);
+      }
+    }
+  };
+
   /**
    * Applies the current subtitle from the playbackManager store
    *
    * It first disables all the VTT and SSA subtitles
-   * It then find the potential index of the applied VTT sub
-   * Or the PlaybackExternalTrack object of the potential SSA sub
-   *
-   * If external and VTT, it shows the correct one
-   * If external and SSA, it loads it in SO
-   *
-   * If embedded, a new transcode is automatically fetched from the playbackManager watchers.
+   * then filters the streams by codec and passes
+   * to the function to apply that codec
    */
   public readonly applyCurrentSubtitle = async (): Promise<void> => {
-    const serverAddress = remote.sdk.api?.basePath;
-    /**
-     * Finding (if it exists) the VTT or SSA track associated to the newly picked subtitle
-     */
-    const vttIdx = playbackManager.currentItemVttParsedSubtitleTracks.findIndex(
-      sub => sub.srcIndex === playbackManager.currentSubtitleStreamIndex
-    );
-    const ass = playbackManager.currentItemAssParsedSubtitleTracks.find(
-      sub => sub.srcIndex === playbackManager.currentSubtitleStreamIndex
-    );
-    const pgs = playbackManager.currentItemPgsParsedSubtitleTracks.find(
-      sub => sub.srcIndex === playbackManager.currentSubtitleStreamIndex
-    );
-    const attachedFonts
-      = playbackManager.currentMediaSource?.MediaAttachments?.filter(a =>
-        this._isSupportedFont(a.MimeType)
-      )
-        .map((a) => {
-          if (a.DeliveryUrl && serverAddress) {
-            return `${serverAddress}${a.DeliveryUrl}`;
-          }
-        })
-        .filter((a): a is string => a !== undefined) ?? [];
-
     if (!mediaElementRef.value) {
       return;
     }
 
-    await nextTick();
-
     /**
-     * Disabling VTT and SSA subs at first
+     * Clear VTT and SSA subs first
      */
     for (const textTrack of mediaElementRef.value.textTracks) {
```

Comment:
Seems this isn't changed in this PR, but typescript reports that textTracks is not iterable (though it might be a false positive because the environment is still not well configured). Can you verify it please?